emsdk/tests/test_memfsmtime.py

```diff
@@ -0,0 +1,78 @@
+import subprocess
+from . import common
+
+
+def test_memfsmtime(tmpdir):
+    with tmpdir.as_cwd():
+        with open("main.c", "w") as f:
+            f.write(
+                r"""\
+#include <assert.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+#include <sys/stat.h>
+
+void mysleep(int delta)
+{
+    time_t end = time(NULL) + delta;
+
+    while (time(NULL)<end);
+}
+
+time_t getmtime(char *fname)
+{
+    struct stat buf;
+
+    stat(fname, &buf);
+    return buf.st_mtime;
+}
+
+void writefile(char *fname, char *content)
+{
+    int fd = open(fname, O_CREAT | O_TRUNC | O_WRONLY, 0777);
+
+    write(fd, content, strlen(content));
+    close(fd);
+}
+
+int main(int argc, char *argv[])
+{
+    char tmpdir[64] = "/tmp/tmpXXXXXXX";
+    char fname[64];
+    time_t t0, t1, t2, t3;
+
+    mkdtemp(tmpdir);
+    strcpy(fname, tmpdir);
+    strcat(fname, "/foo.py");
+    t0 = getmtime(tmpdir);
+    mysleep(1);
+    writefile(fname, "bar = 54\n");
+    t1 = getmtime(tmpdir);
+    t2 = getmtime(fname);
+    assert(t1 > t0);
+    assert(t1 == t2);
+    mysleep(1);
+    unlink(fname);
+    t3 = getmtime(tmpdir);
+    assert(t3 > t1);
+}
+"""
+            )
+
+        subprocess.run(
+            [
+                "emcc",
+                "-s",
+                "MAIN_MODULE=1",
+                "main.c",
+            ],
+            check=True,
+            env=common.env,
+        )
+        out = subprocess.run(
+            ["node", "a.out.js"], capture_output=True, check=False, env=common.env
+        )
+        assert out.returncode == 0
```

Comment:
Why this instead of simply doing `check=True`? I think using `check=True` gives better error messages when it fails.

Reply:
With check = True, I get the following error message (when inserting a dummy assertion inside C code):
```
E               subprocess.CalledProcessError: Command '['node', 'a.out.js']' returned non-zero exit status 1.                                                                                               
                                                                                                                                                                                                             
/usr/local/lib/python3.8/subprocess.py:512: CalledProcessError
```

With check=False and the assertion on out.returncode I can see the assertion error that failed in the C code:
```
>           assert out.returncode == 0                                                                                                                                                                       
E           AssertionError: assert 1 == 0                                                                                                                                                                    
E            +  where 1 = CompletedProcess(args=['node', 'a.out.js'], returncode=1, stdout=b'Assertion failed: 1 == 2, at: main.c,54,main\n', st..._memfsmtime0/a.out.js:32475:5)\n    at runCaller (/tmp/pytest-of-root/pytest-12/test_memfsmtime0/a.out.js:32375:19)\n').returncode                                                                                                                                     
                                                                                                                                                                                                             
emsdk/tests/test_memfsmtime.py:80: AssertionError 
```

Reply:
Hmm I seem to recall having the opposite experience. lgtm then.